### DISCUSSION TITLE
Prompty FileProviders for dotnet

### DISCUSSION THREAD

**Discussion by Tyler-R-Kendrick** [microsoft/semantic-kernel#8680] [Ideas]
Currently, the Prompty implementation relies on System.IO.File for io operations.
Using the Microsoft.Extensions.FileProviders apis would allow for greater extensibility.

More specifically, the current implementation doesn't support loading embedded resources.
Prompts for agents are often embedded into assemblies or loaded from a remote resource; the current implementation doesn't support these client scenarios.

I propose adding an optional file provider that defaults to the physical provider for backwards compatibility.
I also have a PR implementing this request.

**Comment by markwallace-microsoft** [comment]
PR is approved and will be merged shortly
